File: code/preprocess/load_data/data_loader.R
load_hotel_reserve <- function(){
  customer_tb <<- read.csv('data/customer.csv', fileEncoding='UTF-8', header=T, stringsAsFactors=F)
  hotel_tb <<- read.csv('data/hotel.csv', fileEncoding='UTF-8', header=T, stringsAsFactors=F)
  reserve_tb <<- read.csv('data/reserve.csv', fileEncoding='UTF-8', header=T, stringsAsFactors=F)
  
  hotel_tb$is_business <<- hotel_tb$is_business == 'True'
}

load_holiday_mst <- function(){
  holiday_mst <<- read.csv('data/holiday_mst.csv', fileEncoding='UTF-8', header=T, stringsAsFactors=F)
}

load_production <- function(){
  production_tb <<- read.csv('data/production.csv', fileEncoding='UTF-8', header=T, stringsAsFactors=F)
}

load_production_missing_num <- function(){
  production_tb <- read.csv('data/production_missing_num.csv', fileEncoding='UTF-8', header=T, stringsAsFactors=F)
  production_tb$thickness[which(production_tb$thickness=='None')] <- NA
  production_tb$thickness <- as.numeric(production_tb$thickness)
  production_missn_tb <<- production_tb
}

load_production_missing_category <- function(){
  production_tb <- read.csv('data/production_missing_category.csv', fileEncoding='UTF-8', header=T, stringsAsFactors=F)
  production_tb$fault_flg[which(production_tb$fault_flg=='None')] <- NA
  production_tb$fault_flg <- as.logical(production_tb$fault_flg)
  production_missc_tb <<- production_tb
}

load_monthly_index <- function(){
  monthly_index_tb <<- read.csv('data/monthly_index.csv', fileEncoding='UTF-8', header=T, stringsAsFactors=F)
}

load_meros_txt <- function(){
  meros <<- gsub('\t|\n','', paste(readLines('data/txt/meros.txt'), collapse='\n'))
}

load_txt_df <- function(){
  meros <- gsub('\t|\n','', paste(readLines('data/txt/meros.txt'), collapse='\n'))
  aguni <- gsub('\t|\n','', paste(readLines('data/txt/aguni.txt'), collapse='\n'))
  torokko <- gsub('\t|\n','', paste(readLines('data/txt/torokko.txt'), collapse='\n'))
  txt_df <<- data.frame(
    title=c('走れメロス','アグ二の国','トロッコ'),
    text=c(meros,aguni,torokko)
  )
}

File: code/preprocess/load_data/data_loader.py
import pandas as pd
import os


def load_hotel_reserve():
  customer_tb = pd.read_csv(os.path.dirname(__file__)+'/../../data/customer.csv')
  hotel_tb = pd.read_csv(os.path.dirname(__file__) + '/../../data/hotel.csv')
  reserve_tb = pd.read_csv(os.path.dirname(__file__) + '/../../data/reserve.csv')
  return customer_tb, hotel_tb, reserve_tb


def load_holiday_mst():
  holiday_tb = pd.read_csv(os.path.dirname(__file__)+'/../../data/holiday_mst.csv',
                           index_col=False)
  return holiday_tb


def load_production():
  production_tb = pd.read_csv(os.path.dirname(__file__)+'/../../data/production.csv')
  return production_tb


def load_production_missing_num():
  production_tb = pd.read_csv(os.path.dirname(__file__)+'/../../data/production_missing_num.csv')
  return production_tb


def load_production_missing_category():
  production_tb = pd.read_csv(os.path.dirname(__file__)+'/../../data/production_missing_category.csv')
  return production_tb


def load_monthly_index():
  monthly_index_tb = \
    pd.read_csv(os.path.dirname(__file__)+'/../../data/monthly_index.csv')
  return monthly_index_tb


def load_meros_txt():
  with open(os.path.dirname(__file__)+'/../../data/txt/meros.txt', 'r') as f:
    meros = f.read()
    f.close()
  return meros
